db/migrate/20151110191852_create_redirections.rb

```diff
@@ -0,0 +1,10 @@
+class CreateRedirections < ActiveRecord::Migration
+  def change
+    create_table :redirections do |t|
+      t.timestamps null: false
+      t.references :next
```

Comment:
Actually I'd do this for this line:

```ruby
t.references :next, index: true, foreign_key: true, null: false
```

That way we get an index AND a foreign key constraint.